src/extension/executors/github.ts

```diff
@@ -0,0 +1,35 @@
+import { NotebookCellExecution, TextDocument, authentication, window } from 'vscode'
+
+import type { Kernel } from '../kernel'
+import { NotebookCellOutputManager, updateCellMetadata } from '../cell'
+import { OutputType } from '../../constants'
+
+import { parseGitHubURL, getYamlFileContents } from './github/workflows'
+
+export async function github(
+  this: Kernel,
+  exec: NotebookCellExecution,
+  doc: TextDocument,
+  outputs: NotebookCellOutputManager,
+): Promise<boolean> {
+  try {
+    await authentication.getSession('github', ['repo'], { createIfNone: true })
+    const { owner, repo, path, ref } = parseGitHubURL(doc.getText())
+    const json = await getYamlFileContents({ owner, repo, path })
+    await updateCellMetadata(exec.cell, {
```

Comment:
> @degrammer re 2., I think we need to understand what the problem is. I'm uncomfortable merging since it feels like a "bug in the making" to me. Sorry about the delay, I overlooked this ping in the noise of everything else.

I'm fairly certain this is the problem, @degrammer. The `updateCellMetadata` mutates the *input* cell where this metadata should be associated with the github-specific output generated to trigger the renderer. We might want to rename this method to include `Input`. It shouldn't be called unless the source markdown document requires modification.